vignettes/differences-with-python.Rmd

```diff
@@ -0,0 +1,160 @@
+as possible (most of the time switching `.` and `$` to chain methods).
+
+Still, there are a few places where the API diverges. This is often due to
+differences in the language itself. This vignette provides a list of those
+differences.
+
+
+## Converting data between Polars and R
+
+### From R to Polars
+
+The R package provides functions to create polars `DataFrame`, `LazyFrame`, and
+`Series`. Like most of the functions, those are designed to be close to their
+Python counterparts.
+
+Still, R users are more used to `as.*` or `as_*` functions to convert from or to
+other R objects. Therefore, in the documentation, we sometimes prefer using
+`as_polars_df(<data>)` rather than `pl$DataFrame(<data>)`.
+
+### From Polars to R
+
+While Python Polars has `to_pandas()`, we provide methods to convert Polars data
+to standard R objects, such as `$to_list()` or `$to_data_frame()`. However, the
+standard R user might find it more familiar to call `as.data.frame()`, `as.list()`
+or `as.vector()` on Polars structures.
+
+
+## 64-bit integers
+
+R doesn't natively support 64-bit integers (Int64) but this is a completely valid data
+type in Polars, which is based on the Arrow specification. This means that
+handling Int64 values in `polars` objects doesn't deviate from the Python
+setting. However, we need to implement some extra arguments when we want to pass
+data from Polars to R.
+
+In particular, all functions that convert some polars data to R (`as.data.frame()`
+and other methods such as `$to_list()`) have an argument `int64_conversion` which
+specifies how Int64 values should be handled. The default is to convert those
+Int64 to Float64, but it is also possible to convert them to character or to keep
+them as Int64 by using the package `bit64` under the hood.
+
+This option can be set globally using `options(polars.int64_conversion = "<value>")`.
+See `?polars_options()` for more details.
+
+
+## Structs and objects
+
+### Structs
+
+A `Struct` is a data type that is composed of several `Field`s, which themselves
+have a name and a data type.
+
+The way to create a `Series` of type `Struct` is a bit convoluted as it involves
+`data.frame()`, `list()`, and `I()`:
+
+```python
+>>> s = pl.Series([{"a": 1, "b": ["x", "y"]}, {"a": 2, "b": ["z"]}])
+>>> s
+shape: (2,)
+Series: '' [struct[2]]
+[
+        {1,["x", "y"]}
+        {2,["z"]}
+]
+>>> s.dtype
+Struct({'a': Int64, 'b': List(String)})
+```
+
+```{r}
+library(polars)
+
+as_polars_series(
+  data.frame(a = 1:2, b = I(list(c("x", "y"), "z")))
+)
+```
+
+A slightly simpler way would be via `tibble::tibble()` or `data.table::data.table()`:
+
+```{r}
+as_polars_series(
+  tibble::tibble(a = 1:2, b = list(c("x", "y"), "z"))
+)
+```
+
+Finally, one can use the method `$to_struct()` to convert existing columns
+or `Series` to a `Struct`:
+
+```{r}
+x = pl$DataFrame(
+  a = 1:2,
+  b = list(c("x", "y"), "z")
+)
+
+out = x$select(pl$col("a", "b")$to_struct())
+out
+
+out$schema
+```
```

Comment:
I don't see this as something that should be highlighted as a difference from the Python API, just that the data.frame in R is converted to a Struct in Polars?

Reply:
Yes, in the end this can be stored in the docs of `DataType_Struct` instead